### ISSUE TITLE
NS4000 false positive in foreach loop when element used twice

### ISSUE THREAD

**Issue by stukselbax** [nsubstitute/NSubstitute.Analyzers#106]
**NSubstitute** 4.2.0.0
**NSubstitute.Analyzers.CSharp** 1.0.10

When using POCO properties in `foreach` loop and trying to setup two mocks the NS4000 warning occur. When `foreach` loop is removed there are no warnings. Looks like wrong behavior.

```c#
using NSubstitute;
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace Tests
{
    public class Tests
    {
        [Test]
        public void Test1()
        {
            var list = new List<DateTime>(1) { DateTime.Now };
            var enumerator1 = Substitute.For<IEnumerator<DateTime>>();
            var enumerator2 = Substitute.For<IEnumerator<DateTime>>();
            foreach (var date in list)
            {
                // case 1: use date
                enumerator1.Current.Returns(date); // No warnings
                
                // case 2: use date properties
                //enumerator1.Current.Returns(new DateTime(date.Ticks)); // NS4000 here

                enumerator2.Current.Returns(new DateTime(date.Ticks)); // NS4000 here in both cases
            }
        }
    }
}
```

**Comment by tpodolak** [comment]
Hi @stukselbax thanks for reporting. Looks like a bug to me as well. I will try to take a look over the weekend
